Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -113,16 +143,26 @@ String HTMLScriptElement::crossOrigin() const
     return parseCORSSettingsAttribute(attributeWithoutSynchronization(crossoriginAttr));
 }
 
-URL HTMLScriptElement::src() const
+String HTMLScriptElement::src() const
 {
-    return document().completeURL(sourceAttributeValue());
+    return getURLAttributeForBindings(WebCore::HTMLNames::srcAttr).string();
```

Comment:
Yes, seems like you did exactly the right thing here. The only better thing would be if there was a way we could get the getter to be generated by the `[Reflect]` attribute, and still hand-write the setter. You might think we could do it with `[Reflect, CustomSetter]` but sadly a custom setter is too “custom” for what we want.

Moving the code inside `addSubresourceAttributeURLs` was exactly the right thing to do.

I agree that it’s unlikely `src()` is used anywhere else. I suggest you do a grep as well as counting on `URL` vs. `String`, since there are some types of super-simplistic calls that might still compile with the wrong type, so nice to be thorough.

Reply:
I've had a look through and I can't seem to find a usage anywhere.